File: Testing/Code for Testing/Basic/JS/T401_control/original.c
#include <stdio.h>

int classify(int n) {
    if (n > 0) {
        return 1;
    } else if (n < 0) {
        return -1;
    } else {
        return 0;
    }
}

int factorial(int n) {
    int result = 1;
    int i = 1;
    while (i <= n) {
        result = result * i;
        i = i + 1;
    }
    return result;
}

int fibonacci(int n) {
    if (n <= 1) {
        return n;
    }
    int a = 0;
    int b = 1;
    int i = 2;
    while (i <= n) {
        int temp = a + b;
        a = b;
        b = temp;
        i = i + 1;
    }
    return b;
}

int sum_range(int lo, int hi) {
    int total = 0;
    int i = lo;
    while (i <= hi) {
        total = total + i;
        i = i + 1;
    }
    return total;
}

int power(int base, int exp) {
    int result = 1;
    int i = 0;
    while (i < exp) {
        result = result * base;
        i = i + 1;
    }
    return result;
}

int gcd(int a, int b) {
    while (b != 0) {
        int temp = b;
        b = a % b;
        a = temp;
    }
    return a;
}

int main() {
    printf("classify 5: %d\n", classify(5));
    printf("classify -3: %d\n", classify(-3));
    printf("classify 0: %d\n", classify(0));
    printf("fact 0: %d\n", factorial(0));
    printf("fact 5: %d\n", factorial(5));
    printf("fact 10: %d\n", factorial(10));
    printf("fib 0: %d\n", fibonacci(0));
    printf("fib 1: %d\n", fibonacci(1));
    printf("fib 10: %d\n", fibonacci(10));
    printf("fib 15: %d\n", fibonacci(15));
    printf("sum 1..10: %d\n", sum_range(1, 10));
    printf("sum 1..100: %d\n", sum_range(1, 100));
    printf("pow 2^10: %d\n", power(2, 10));
    printf("pow 3^5: %d\n", power(3, 5));
    printf("gcd 48 18: %d\n", gcd(48, 18));
    printf("gcd 100 75: %d\n", gcd(100, 75));
    return 0;
}

File: Testing/Code for Testing/Intermediate/Python/T301_calculator/original.py
"""Simple integer calculator with named operations."""


def add(a: int, b: int) -> int:
    return a + b


def subtract(a: int, b: int) -> int:
    return a - b


def multiply(a: int, b: int) -> int:
    return a * b


def safe_divide(a: int, b: int) -> int:
    if b == 0:
        return -1
    return a // b


def power(base: int, exp: int) -> int:
    result: int = 1
    i: int = 0
    while i < exp:
        result = result * base
        i = i + 1
    return result


def factorial(n: int) -> int:
    if n <= 1:
        return 1
    return n * factorial(n - 1)


def gcd(a: int, b: int) -> int:
    while b != 0:
        temp: int = b
        b = a % b
        a = temp
    return a


def main() -> int:
    print(add(10, 25))
    print(subtract(100, 37))
    print(multiply(6, 7))
    print(safe_divide(100, 3))
    print(safe_divide(10, 0))
    print(power(2, 10))
    print(factorial(10))
    print(gcd(48, 18))
    print(gcd(100, 75))

    # Compose operations
    result: int = add(multiply(3, 4), subtract(20, 8))
    print(result)
    return 0


if __name__ == "__main__":
    main()
